project3_companythema_dag_v5.py

```diff
@@ -0,0 +1,156 @@
+    return conn.cursor()
+
+def read_csv_and_yf(bucket_name: str, key:str) -> pd.DataFrame:
+    s3_client = get_s3_connection()
+    obj = s3_client.get_object(Bucket=bucket_name, Key=key)
+    body = obj['Body'].read().decode('utf-8')
+    df = pd.read_csv(io.StringIO(body))
+    end_date = datetime.now().date()
+    start_date = end_date - timedelta(days=7)
+    stock=yf.download(df['Symbol'].tolist(), start=start_date, end=end_date)
+    df_v = stock['Volume'].reset_index()
+    df_new = df_v.melt(id_vars='Date',var_name='ticker',value_name='volume')
+
+    csv_buffer = StringIO()
+    df_new.to_csv(csv_buffer, index=False)
+
+    s3_client.put_object(
+        Bucket=bucket_name,
+        Key='volume_data.csv',
+        Body=csv_buffer.getvalue()
+    )
+
+    return 'volume_data.csv'
+
+def create_volume_to_redshift():
+    cursor = get_Redshift_connection()
+    # 테이블 생성 (이미 존재할 경우 무시)
+    create_table_query = f"""
+    CREATE TABLE IF NOT EXISTS raw_data.volume (
+        Date TIMESTAMP,
+        Symbol VARCHAR(200),
+        Volume BIGINT 
+    );
+    """
+    cursor.execute(create_table_query)
+
+    return True
+
+with DAG(
+    dag_id='project3_companythema_dag_v5',
+    schedule_interval='0 0 * * *', # 매일0시 0분
+    start_date=datetime(2024,11,1),
+    catchup=False,
+    default_args = {
+        'retries': 1
+    }
+) as dag:
+    
+    create_table_redshift = PostgresOperator(
+        task_id='create_table_redshift',
+        postgres_conn_id='redshift_dev_db_project',
+        sql="""
+        CREATE TABLE IF NOT EXISTS raw_data.nasdaq_ticker (
+            Symbol VARCHAR(200),
+            Name VARCHAR(200),
+            Last_Sale VARCHAR(200),
+            Net_Change FLOAT,
+            Percent_Change FLOAT,
+            Market_Cap FLOAT,
+            Country VARCHAR(200),
+            IPO_Year INT,
+            Volume BIGINT,
+            Sector VARCHAR(200),
+            Industry VARCHAR(200)
+        );
+        """
+    )
+
+    load_data_s3_to_red = S3ToRedshiftOperator(
+        task_id='load_data_s3_to_red',
+        schema='raw_data',
+        table='nasdaq_ticker',
+        s3_bucket='project3-team8-bucket-bdj',
+        s3_key='modified_nasdaq_ticker.csv',
+        copy_options=['CSV','IGNOREHEADER 1'],
+        redshift_conn_id='redshift_dev_db_project',
+        aws_conn_id='project_s3_conn',
+        method='REPLACE'
+    )
+
+    read_csv_and_yf_task = PythonOperator(
+        dag=dag,
+        task_id='read_csv_and_yf',
+        python_callable=read_csv_and_yf,
+        op_kwargs={
+            'bucket_name': 'project3-team8-bucket-bdj',
+            'key': 'modified_nasdaq_ticker.csv'
+        },
+        provide_context=True
+    )
+
+    create_volume_to_redshift_task = PythonOperator(
+        dag=dag,
+        task_id='create_volume',
+        python_callable=create_volume_to_redshift
+    )
+
+    load_data_s3_to_red2 = S3ToRedshiftOperator(
+        task_id='load_data_s3_to_red2',

```

Comment:
task_id를 `_2` 가 아니고 table name을 같이 쓰는건 어떨까요?
```suggestion
        task_id='load_data_s3_to_red_volume',

```